src/components/create_cluster/node_count_selector.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+
+import React from 'react';
+import { connect } from 'react-redux';
+import NumberPicker from './number_picker.js';
+import PropTypes from 'prop-types';
+
+// NodeCountSelector is a component that allows a user to pick a number of
+// nodes by incrementing / decrementing a node count value or adjusting min and
+// max values for scaling limits.
+//
+// State explanation:
+//
+// autoscalingEnabled - Boolean whether cluster supports autoscaling or not.
+//                      This is used to determine whether only number of nodes
+//                      is displayed of if there is pickers for min/max values.
+//
+// scaling - The current value of the input field[s].
+//
+
+class NodeCountSelector extends React.Component {
+  state = {
+    autoscalingEnabled: this.props.autoscalingEnabled,
+    scaling: this.props.scaling,
+  };
+
+  updateScalingMin = numberPicker => {
+    this.setState({
+      scaling: {
+        automatic: this.state.scaling.automatic,
+        min: numberPicker.value,
+        minValid: numberPicker.valid,
+        max: this.state.scaling.max,
+        maxValid: this.state.scaling.maxValid,
+      },
+    });
+  };
+
+  updateScalingMax = numberPicker => {
+    this.setState({
+      scaling: {
+        automatic: this.state.scaling.automatic,
+        min: this.state.scaling.min,
+        minValid: this.state.scaling.minValid,
+        max: numberPicker.value,
+        maxValid: numberPicker.valid,
+      },
+    });
+  };
+
+  updateNodeCount = numberPicker => {
+    this.setState({
+      scaling: {
+        automatic: false,
```

Comment:
What about aligning this? I know `updateNodeCount` it is only used  when it is false but it could lead to mistakes in the future
```suggestion
        automatic: this.state.scaling.automatic,
```

Reply:
Oh, that's leftover. It should be actually removed. Thanks for pointing it out!